src/OrchardCore.Modules/OrchardCore.Sms/Drivers/SmsSettingsDisplayDriver.cs

```diff
@@ -55,14 +53,14 @@ public override async Task<IDisplayResult> EditAsync(SmsSettings settings, Build
             .Select(provider => new SelectListItem(provider, provider))
             .ToArray();
         }).Location("Content:1")
-        .OnGroup(GroupId);
+        .OnGroup(SmsConstants.SettingsGroupId);
```

Comment:
I rather have it is constants because it is a value that should be used by all implementation. Implementation don't need to do a hard reference to the SMS module. I know you can simply type "sms". But I thought it would be better to have a reusable constant.

The twilio does not need to be atomic I just did it this way so I don't have to initialize the client multiple times is the same request.

Reply:
I'm not against constants but how they are defined, for example I prefer `SmsSettings.GroupId`, `SmsProvider.Twilio` and so on.

Just saw that `EnsureClientIsIsInitializedAsync()` updates different fields of the twilio provider which is a singleton.